shared/pkg/utils/image_helper/image_helper.go

```diff
@@ -225,14 +225,14 @@ func GetImageLayerCommands(imageName string, sharedConf *sharedconfig.Config) ([
 	return layerCommands, nil
 }
 
-func GetHashFromRepoOrManifestDigest(repoDigests []string, manifestDigest string, imageName string) string {
+func GetHashFromRepoDigestsOrImageID(repoDigests []string, imageID string, imageName string) string {
```

Comment:
I wonder if this function should handle the error when len(repoDigests) == 0 and imageID == "" ? Then we could have a common error from all the scanners if we don't have the repoHash or imageID. Right now if passed imageID == "" then it'll return "" not a hash or an error.

Reply:
sure, make sense since we are counting on the existing of the hash - I will update the function.

Reply:
I've updated the function [here](https://github.com/openclarity/kubeclarity/pull/510/commits/919aeb1f4d9932d83080c8f5129d0864e1650fca), BUT I kept the old behavior of the scanners for now to basically ignore a missing hash - I've decided just to print a warning and if that will appear we will need to handle it (didn't wanted to break the current the behavior too much)
I think that it's more important when generating the sbom (and this is why we check it before) - but again, maybe we will need to also return error in the scanners flow in the future.
WDYT?

Reply:
Yeah happy with that, if we run into more breakages then we can address it then